videoframegrid.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from PyQt5 import QtWidgets
+from coordinates import VideoFrameCoordinates
+from videoframes import LiveVideoFrame
+
+
+class VideoFrameGrid(QtWidgets.QGridLayout):
+    """The VideoFrameGrid is the GridLayout container for all LiveVideoFrames.
+
+    The VideoFrameGrid keeps track of the position of all LiveVideoFrames and
+    is in charge of reparenting a VideoFrame back to it's original position
+    after a toggled fullscreen mode.
+    """
+
+    def __init__(self):
+        """Creates a new VideoFrameGrid.
+
+        The VidoeFrameGrid holds a list of the active videoframe objects as
+        well as a coordinates object to keep track of the current grid
+        coordinates (beginning at (0, 0)).
+        """
+        super(VideoFrameGrid, self).__init__()
+        self.videoframes = []
+        self.coordinates = VideoFrameCoordinates(x=0, y=0)
+
+    def _add_videoframe(self, videoframe):
+        """Adds the provided videoframeobject to the VideoFrameGrid."""
+        self.videoframes.append(videoframe)
+        self.addWidget(videoframe, self.coordinates.x, self.coordinates.y)
+        self.coordinates = self.coordinates.update_coordinates()
+
+    def _create_videoframe(self, vlc_instance, stream_options, quality):
+        """Creates a new LiveVideoFrame object."""
+        return LiveVideoFrame(vlc_instance, stream_options, quality)
+
+    def add_new_videoframe(self, vlc_instance, stream_options, quality, move_handle):
+        """Creates and adds a new LiveVideoFrame to the VideoFrameGrid."""
+        videoframe = self._create_videoframe(vlc_instance, stream_options, quality)
+        videoframe._move = move_handle
```

Comment:
IMO I'm not super stoked about the `self.move_frame` function being passed down to the individual videoframes since that feels a bit like backwards architecture (was so before the PR), but perhaps that's a thing for a later PR.